Amazon.QLDB.Driver/session/BaseSessionPool.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver
+{
+    using System.Threading;
+
+    /// <summary>
+    /// Base class for Session Pool.
+    /// </summary>
+    internal abstract class BaseSessionPool
```

Comment:
There's a lot of shared initialization of class fields that can belong in a constructor here.

Reply:
Done.